thoth-app/src/component/work_status_modal.rs

```diff
@@ -0,0 +1,116 @@
+use crate::string::CANCEL_BUTTON;
+use crate::string::SAVE_BUTTON;
+use thoth_api::account::model::AccountDetails;
+use yew::html;
+use yew::prelude::*;
+
+pub struct ConfirmWorkStatusComponent {
+    show: bool,
+}
+
+#[derive(PartialEq, Properties)]
+pub struct Props {
+    pub onsubmit: Callback<()>,
+    pub object_name: String,
+    pub current_user: AccountDetails,
+    pub current_state_unpublished: bool,
+    pub is_published: bool,
+    #[prop_or_default]
+    pub deactivated: bool,
+}
+
+pub enum Msg {
+    ToggleConfirmWorkStatusDisplay(bool),
+    CloseModalAndUpdateWork,
+}
+
+impl Component for ConfirmWorkStatusComponent {
+    type Message = Msg;
+    type Properties = Props;
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        ConfirmWorkStatusComponent { show: false }
+    }
+
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Msg::ToggleConfirmWorkStatusDisplay(value) => {
+                self.show = value;
+                true
+            }
+            Msg::CloseModalAndUpdateWork => {
```

Comment:
This variant isn't actually updating anything, so let's just call it `CloseModal`